pkg/volume/aws_ebs/aws_ebs.go

```diff
@@ -241,6 +243,232 @@ func (plugin *awsElasticBlockStorePlugin) ConstructVolumeSpec(volName, mountPath
+	// Note: in this case podUID is always empty
+	if strings.Contains(volName, awsElasticBlockStorePluginName) {
+		elems := strings.Split(volName, "/")
+		vID := elems[len(elems)-1]
+
+		if !strings.Contains(vID, "vol-") {
+			return "", fmt.Errorf("failed to retreive volume id got %s from volName: %s", vID, volName)
+		}
+		return vID, nil
+	}
+
+	// If podUID is empty, then we have to fail at this point and not try to
+	// reconstruct volumeID from globalMapPath
+	if len(podUID) == 0 {
+		return "", fmt.Errorf("failed to retreive volume id  podUID is empty for volName: %s", volName)
+	}
+
+	// Otherwise, check that we have a normal volName and determine volumeID
+	// by reading volumeID directory, matching on next level which should be volumeName
+	// and then finally matching on one more level down the tree which will be podUID
+	files, err := ioutil.ReadDir(pluginDir)
+	if err != nil {
+		return "", fmt.Errorf("failed to retreive volume id from globalMapPath based on volName and podUID: %s - %s", volName, podUID)
+	}
+	for _, f := range files {
+		newPath := filepath.Join(pluginDir, f.Name())
+		vols, err := ioutil.ReadDir(newPath)
+		if err != nil {
+			return "", fmt.Errorf("failed to retreive volume id from globalMapPath based on volName and podUID: %s - %s from path %s - %v", volName, podUID, newPath, err)
+		}
+		for _, v := range vols {
+			// we have a volume ID now check next level for volName match
+			vName := v.Name()
+			if vName == volName {
+				// we have a match on volName now check next level for podUID
+				newPath2 := filepath.Join(newPath, v.Name())
+				podsuid, err := ioutil.ReadDir(newPath2)
+				if err != nil {
+					return "", fmt.Errorf("failed to retreive volume id from globalMapPath based on volName and podUID: %s - %s from path %s - %v", volName, podUID, newPath, err)
+				}
+				for _, n := range podsuid {
+					pID := n.Name()
+					if pID == podUID {
+						// we found it
+						return f.Name(), nil
+					}
+				}
+			}
+		}
+	}
+	return "", fmt.Errorf("failed to retreive volume id from globalMapPath based on volName and podUID: %s - %s", volName, podUID)
+}
+
+func (c *awsElasticBlockStoreUnmapper) TearDownDevice(mapPath, devicePath string) error {
+	return nil
+}
+
+type awsElasticBlockStoreUnmapper struct {
+	*awsElasticBlockStore
+}
+
+var _ volume.BlockVolumeUnmapper = &awsElasticBlockStoreUnmapper{}
+
+type awsElasticBlockStoreMapper struct {
+	*awsElasticBlockStore
+	fsType      string
+	readOnly    bool
+	diskMounter *mount.SafeFormatAndMount
+}
+
+var _ volume.BlockVolumeMapper = &awsElasticBlockStoreMapper{}
+
+func (b *awsElasticBlockStoreMapper) SetUpDevice() (string, error) {
+	return "", nil
+}
+
+// GetGlobalMapPath returns global map path and error
+// path: plugins/kubernetes.io/{PluginName}/volumeDevices/volumeID/volName/
+//       plugins/kubernetes.io/aws-ebs/volumeDevices/vol-XXXXXX/myebsvolume/
+func (ebs *awsElasticBlockStore) GetGlobalMapPath(spec *volume.Spec) (string, error) {
```

Comment:
@screeley44 

From my debug result, global mount path(filesystem volume) is corresponding to provisioning type.
- Static provisioning
```
/var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/vol-079033653d3945a51
```
In this case, volumeID is `vol-0b2360c4480ad10f4`

- Dynamic provisioning
```
/var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/aws/us-east-2c/vol-0b2360c4480ad10f4
```
In this case, volumeID is `aws/us-east-2c/vol-0b2360c4480ad10f4`.
Provisioner add `aws` and `volume region` into volumeID.

But either case, the global mount path is `/var/lib/kubelet/plugins/kubernetes.io/aws-ebs/mounts/{volumeID}`.

Therefore, we should use same format for Block volume case.
Supposed global map path for Block volume is;
- /var/lib/kubelet/plugins/kubernetes.io/aws-ebs/`volumeDevices`/{volumeID}

And then under `{volumeID}` dir we will have symbolic links whose name is pod's UUID.
So this method should be like this;
```
 func (ebs *awsElasticBlockStore) GetGlobalMapPath(spec *volume.Spec) (string, error) {
       volumeSource, _, err := getVolumeSource(spec)
       if err != nil {
               return "", err
       }
       return path.Join(ebs.plugin.host.GetVolumeDevicePluginDir(awsElasticBlockStorePluginName), string(volumeSource.VolumeID)), nil
 }
```

Reply:
ack - backed out all the volName changes and fixed GetGlobalMapMapth based on spec volumeID.